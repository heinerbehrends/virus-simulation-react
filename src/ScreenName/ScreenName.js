import React, { useState } from 'react';
import PropTypes from 'prop-types';
import {
  FormContainer, Form, FormGroup, HiddenLabel, ButtonForm, TextInput,
} from './ScreenNameStyled';

const ScreenName = ({ sendName, screenName }) => {
  const [input, setInput] = useState('');

  const handleChange = ({ target: { value } }) => setInput(value);

  const sendScreenName = (event) => {
    event.preventDefault();
    if (!input) {
      return;
    }
    sendName(input);
  };

  return (
    screenName.length ? null : (
      <FormContainer>
        <Form
          id="screenName"
          name="screenName"
          autoComplete="off"
          onSubmit={sendScreenName}
        >
          <HiddenLabel htmlFor="enter-screen-name">Enter screen name</HiddenLabel>
          <FormGroup>
            <TextInput
              placeholder="Enter screen name to start"
              onChange={handleChange}
              type="text"
              id="enter-screen-name"
              name="screenName"
              autoFocus
            />
            <ButtonForm type="submit" value="Go" />
          </FormGroup>
        </Form>
      </FormContainer>
    )
  );
};

ScreenName.propTypes = {
  sendName: PropTypes.func.isRequired,
  screenName: PropTypes.string.isRequired,
};

export default ScreenName;
