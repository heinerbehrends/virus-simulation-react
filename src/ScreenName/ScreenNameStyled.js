import styled from 'styled-components';
import { InputStyled, ButtonStyled } from '../Button/ButtonStyled';
import * as style from '../styleConstants';

export const FormContainer = styled.div`
  display: flex;
  justify-content: center;
  box-sizing: border-box;
`;

export const Form = styled.form`
  display: flex;
  flex-flow: row wrap;
  align-items: center;
  -webkit-box-orient: horizontal;
  -webkit-box-direction: normal;
`;

export const HiddenLabel = styled.label`
  display: none!important;
`;

export const FormGroup = styled.div`
  flex: 0 0 auto;
  -webkit-box-orient: horizontal;
  -webkit-box-direction: normal;
  -webkit-box-flex: 0;
`;

export const ButtonForm = styled(ButtonStyled)`
  border-radius: 0 .3rem .3rem 0;
`;

export const TextInput = styled(InputStyled)`
  text-align: center;
  ${'' /* border-radius: .3rem 0 0 .3rem; */}
  ::placeholder {
    color: ${style.textColor};
  }
  :focus {
    ::placeholder {
      color: ${style.fadedText};
    }
  }
`;
