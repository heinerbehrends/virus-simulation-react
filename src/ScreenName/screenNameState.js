import { take } from 'redux-saga/effects';

export const SEND_NAME = 'SEND_NAME';

export const sendNameAction = name => ({
  type: SEND_NAME,
  name,
});

export const screenNameReducer = (state = '', action) => {
  switch (action.type) {
    case SEND_NAME:
      return action.name;
    default:  
      return state;
  }    
};  

export const getScreenName = state => state.screenName;

export function* sendNameIO(socket) {
  const action = yield take(SEND_NAME);
  const { name } = action;
  socket.emit('sendName', { name });
}
