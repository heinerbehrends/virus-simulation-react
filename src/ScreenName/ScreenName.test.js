import React from 'react';
import { shallow } from 'enzyme';
import ScreenName from './ScreenName';
import 'jest-styled-components';

describe('<ScreenName />', () => {
  it('renders a form with id screenName when the screenName prop is an empty string', () => {
    const mockCallback = jest.fn();
    const wrapper = shallow((<ScreenName sendName={mockCallback} screenName="" />));
    expect(wrapper.find('#screenName').exists()).toBe(true);
  });
  it('renders nothing if screenName is not an empty string', () => {
    const mockCallback = jest.fn();
    const wrapper = shallow((<ScreenName sendName={mockCallback} screenName="pizarro" />));
    expect(wrapper.type()).toEqual(null);
  });
  it('calls the provided function on submit if state.screenName is not an empty string', () => {
    const mockCallback = jest.fn();
    const mockEvent = { preventDefault: () => {} };
    const wrapper = shallow((<ScreenName sendName={mockCallback} screenName="" />));
    wrapper.setState({ screenName: 'pizarro' });
    wrapper.find('#screenName').simulate('submit', mockEvent);
    expect(mockCallback).toHaveBeenCalledTimes(1);
  });
  it('calls the provided function on submit with the correct argument', () => {
    const mockCallback = jest.fn();
    const mockEvent = { preventDefault: () => {} };
    const wrapper = shallow((<ScreenName sendName={mockCallback} screenName="" />));
    wrapper.setState({ screenName: 'pizarro' });
    wrapper.find('#screenName').simulate('submit', mockEvent);
    expect(mockCallback).toHaveBeenCalledWith('pizarro');
  });
  it('does not call the provided function on submit if state.screenName is an empty string', () => {
    const mockCallback = jest.fn();
    const mockEvent = { preventDefault: () => {} };
    const wrapper = shallow((<ScreenName sendName={mockCallback} screenName="" />));
    wrapper.find('#screenName').simulate('submit', mockEvent);
    expect(mockCallback).toHaveBeenCalledTimes(0);
  });

  it('matches the snapshot', () => {
    const mockCallback = jest.fn();
    const wrapper = shallow(<ScreenName sendName={mockCallback} screenName="" />);
    expect(wrapper).toMatchSnapshot();
  });
});
