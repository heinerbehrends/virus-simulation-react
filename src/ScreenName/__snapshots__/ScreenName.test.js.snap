// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ScreenName /> matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ScreenName
    screenName=""
    sendName={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <ForwardRef
        autoComplete="off"
        id="screenName"
        name="screenName"
        onSubmit={[Function]}
      >
        <ForwardRef
          htmlFor="enter-screen-name"
        >
          Enter screen name
        </ForwardRef>
        <ForwardRef>
          <ForwardRef
            autoFocus={true}
            id="enter-screen-name"
            name="screenName"
            onChange={[Function]}
            placeholder="Enter screen name to start"
            type="text"
          />
          <ForwardRef
            type="submit"
            value="Go"
          />
        </ForwardRef>
      </ForwardRef>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "autoComplete": "off",
        "children": Array [
          <ForwardRef
            htmlFor="enter-screen-name"
          >
            Enter screen name
          </ForwardRef>,
          <ForwardRef>
            <ForwardRef
              autoFocus={true}
              id="enter-screen-name"
              name="screenName"
              onChange={[Function]}
              placeholder="Enter screen name to start"
              type="text"
            />
            <ForwardRef
              type="submit"
              value="Go"
            />
          </ForwardRef>,
        ],
        "id": "screenName",
        "name": "screenName",
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Enter screen name",
            "htmlFor": "enter-screen-name",
          },
          "ref": null,
          "rendered": "Enter screen name",
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-ifAKCX",
              "isStatic": true,
              "rules": Array [
                "
  display: none!important;
",
              ],
            },
            "displayName": "styled.label",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-ifAKCX",
            "target": "label",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <ForwardRef
                autoFocus={true}
                id="enter-screen-name"
                name="screenName"
                onChange={[Function]}
                placeholder="Enter screen name to start"
                type="text"
              />,
              <ForwardRef
                type="submit"
                value="Go"
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "autoFocus": true,
                "id": "enter-screen-name",
                "name": "screenName",
                "onChange": [Function],
                "placeholder": "Enter screen name to start",
                "type": "text",
              },
              "ref": null,
              "rendered": null,
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-gzVnrw",
                  "isStatic": true,
                  "rules": Array [
                    "
  display: inline-block;
  vertical-align: middle;
  font-size: ",
                    "4vw",
                    ";
  font-weight: ",
                    "300",
                    ";
  padding: 3vw 6vw;
  margin-bottom: ",
                    "9vw",
                    ";
  color: ",
                    "#7A828A",
                    ";
  border: .5px solid ",
                    "#CED4DA",
                    ";
  background-color: ",
                    "#FFF",
                    ";
  box-shadow: ",
                    "1.8px 2.4px 1.8px rgba(0, 0, 0, 0.05)",
                    ";
  @media screen and (min-width: 445px) {
    margin-bottom: ",
                    "3rem",
                    ";
    font-size: 1.1rem;
    padding: ",
                    "0.8rem 1.6rem",
                    ";
  };
  :focus {
    outline: 0;
    border: 1px solid ",
                    "#BDEAFE",
                    ";
  }
",
                    "
  text-align: left;
  border-radius: .3rem 0 0 .3rem;
  ::placeholder {
    color: ",
                    "#7A828A",
                    ";
  }
  :focus {
    ::placeholder {
      color: ",
                    "#AAB0B5",
                    ";
    }
  }
",
                  ],
                },
                "displayName": "Styled(styled.input)",
                "foldedComponentIds": Array [
                  "sc-bdVaJa",
                ],
                "render": [Function],
                "styledComponentId": "sc-gzVnrw",
                "target": "input",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              },
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "type": "submit",
                "value": "Go",
              },
              "ref": null,
              "rendered": null,
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bZQynM",
                  "isStatic": true,
                  "rules": Array [
                    "
  display: inline-block;
  vertical-align: middle;
  font-size: ",
                    "4vw",
                    ";
  font-weight: ",
                    "300",
                    ";
  padding: 3vw 6vw;
  margin-bottom: ",
                    "9vw",
                    ";
  color: ",
                    "#7A828A",
                    ";
  border: .5px solid ",
                    "#CED4DA",
                    ";
  background-color: ",
                    "#FFF",
                    ";
  box-shadow: ",
                    "1.8px 2.4px 1.8px rgba(0, 0, 0, 0.05)",
                    ";
  @media screen and (min-width: 445px) {
    margin-bottom: ",
                    "3rem",
                    ";
    font-size: 1.1rem;
    padding: ",
                    "0.8rem 1.6rem",
                    ";
  };
  :focus {
    outline: 0;
    border: 1px solid ",
                    "#BDEAFE",
                    ";
  }
",
                    "
  border-radius: .3rem;
  text-align: center;
  :active {
    box-shadow: inset ",
                    "1.8px 2.4px 1.8px rgba(0, 0, 0, 0.05)",
                    ";
  }
",
                    "
  border-radius: 0 .3rem .3rem 0;
",
                  ],
                },
                "displayName": "Styled(Styled(styled.input))",
                "foldedComponentIds": Array [
                  "sc-bdVaJa",
                  "sc-bwzfXH",
                ],
                "render": [Function],
                "styledComponentId": "sc-bZQynM",
                "target": "input",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              },
            },
          ],
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-EHOje",
              "isStatic": true,
              "rules": Array [
                "
  flex: 0 0 auto;
  -webkit-box-orient: horizontal;
  -webkit-box-direction: normal;
  -webkit-box-flex: 0;
",
              ],
            },
            "displayName": "styled.div",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-EHOje",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bxivhb",
          "isStatic": true,
          "rules": Array [
            "
  display: flex;
  flex-flow: row wrap;
  align-items: center;
  -webkit-box-orient: horizontal;
  -webkit-box-direction: normal;
",
          ],
        },
        "displayName": "styled.form",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bxivhb",
        "target": "form",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-htpNat",
        "isStatic": true,
        "rules": Array [
          "
  display: flex;
  justify-content: center;
  box-sizing: border-box;
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-htpNat",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <ForwardRef
          autoComplete="off"
          id="screenName"
          name="screenName"
          onSubmit={[Function]}
        >
          <ForwardRef
            htmlFor="enter-screen-name"
          >
            Enter screen name
          </ForwardRef>
          <ForwardRef>
            <ForwardRef
              autoFocus={true}
              id="enter-screen-name"
              name="screenName"
              onChange={[Function]}
              placeholder="Enter screen name to start"
              type="text"
            />
            <ForwardRef
              type="submit"
              value="Go"
            />
          </ForwardRef>
        </ForwardRef>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "autoComplete": "off",
          "children": Array [
            <ForwardRef
              htmlFor="enter-screen-name"
            >
              Enter screen name
            </ForwardRef>,
            <ForwardRef>
              <ForwardRef
                autoFocus={true}
                id="enter-screen-name"
                name="screenName"
                onChange={[Function]}
                placeholder="Enter screen name to start"
                type="text"
              />
              <ForwardRef
                type="submit"
                value="Go"
              />
            </ForwardRef>,
          ],
          "id": "screenName",
          "name": "screenName",
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Enter screen name",
              "htmlFor": "enter-screen-name",
            },
            "ref": null,
            "rendered": "Enter screen name",
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "sc-ifAKCX",
                "isStatic": true,
                "rules": Array [
                  "
  display: none!important;
",
                ],
              },
              "displayName": "styled.label",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "sc-ifAKCX",
              "target": "label",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            },
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <ForwardRef
                  autoFocus={true}
                  id="enter-screen-name"
                  name="screenName"
                  onChange={[Function]}
                  placeholder="Enter screen name to start"
                  type="text"
                />,
                <ForwardRef
                  type="submit"
                  value="Go"
                />,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "autoFocus": true,
                  "id": "enter-screen-name",
                  "name": "screenName",
                  "onChange": [Function],
                  "placeholder": "Enter screen name to start",
                  "type": "text",
                },
                "ref": null,
                "rendered": null,
                "type": Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-gzVnrw",
                    "isStatic": true,
                    "rules": Array [
                      "
  display: inline-block;
  vertical-align: middle;
  font-size: ",
                      "4vw",
                      ";
  font-weight: ",
                      "300",
                      ";
  padding: 3vw 6vw;
  margin-bottom: ",
                      "9vw",
                      ";
  color: ",
                      "#7A828A",
                      ";
  border: .5px solid ",
                      "#CED4DA",
                      ";
  background-color: ",
                      "#FFF",
                      ";
  box-shadow: ",
                      "1.8px 2.4px 1.8px rgba(0, 0, 0, 0.05)",
                      ";
  @media screen and (min-width: 445px) {
    margin-bottom: ",
                      "3rem",
                      ";
    font-size: 1.1rem;
    padding: ",
                      "0.8rem 1.6rem",
                      ";
  };
  :focus {
    outline: 0;
    border: 1px solid ",
                      "#BDEAFE",
                      ";
  }
",
                      "
  text-align: left;
  border-radius: .3rem 0 0 .3rem;
  ::placeholder {
    color: ",
                      "#7A828A",
                      ";
  }
  :focus {
    ::placeholder {
      color: ",
                      "#AAB0B5",
                      ";
    }
  }
",
                    ],
                  },
                  "displayName": "Styled(styled.input)",
                  "foldedComponentIds": Array [
                    "sc-bdVaJa",
                  ],
                  "render": [Function],
                  "styledComponentId": "sc-gzVnrw",
                  "target": "input",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                },
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "type": "submit",
                  "value": "Go",
                },
                "ref": null,
                "rendered": null,
                "type": Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-bZQynM",
                    "isStatic": true,
                    "rules": Array [
                      "
  display: inline-block;
  vertical-align: middle;
  font-size: ",
                      "4vw",
                      ";
  font-weight: ",
                      "300",
                      ";
  padding: 3vw 6vw;
  margin-bottom: ",
                      "9vw",
                      ";
  color: ",
                      "#7A828A",
                      ";
  border: .5px solid ",
                      "#CED4DA",
                      ";
  background-color: ",
                      "#FFF",
                      ";
  box-shadow: ",
                      "1.8px 2.4px 1.8px rgba(0, 0, 0, 0.05)",
                      ";
  @media screen and (min-width: 445px) {
    margin-bottom: ",
                      "3rem",
                      ";
    font-size: 1.1rem;
    padding: ",
                      "0.8rem 1.6rem",
                      ";
  };
  :focus {
    outline: 0;
    border: 1px solid ",
                      "#BDEAFE",
                      ";
  }
",
                      "
  border-radius: .3rem;
  text-align: center;
  :active {
    box-shadow: inset ",
                      "1.8px 2.4px 1.8px rgba(0, 0, 0, 0.05)",
                      ";
  }
",
                      "
  border-radius: 0 .3rem .3rem 0;
",
                    ],
                  },
                  "displayName": "Styled(Styled(styled.input))",
                  "foldedComponentIds": Array [
                    "sc-bdVaJa",
                    "sc-bwzfXH",
                  ],
                  "render": [Function],
                  "styledComponentId": "sc-bZQynM",
                  "target": "input",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                },
              },
            ],
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "sc-EHOje",
                "isStatic": true,
                "rules": Array [
                  "
  flex: 0 0 auto;
  -webkit-box-orient: horizontal;
  -webkit-box-direction: normal;
  -webkit-box-flex: 0;
",
                ],
              },
              "displayName": "styled.div",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "sc-EHOje",
              "target": "div",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            },
          },
        ],
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bxivhb",
            "isStatic": true,
            "rules": Array [
              "
  display: flex;
  flex-flow: row wrap;
  align-items: center;
  -webkit-box-orient: horizontal;
  -webkit-box-direction: normal;
",
            ],
          },
          "displayName": "styled.form",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bxivhb",
          "target": "form",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-htpNat",
          "isStatic": true,
          "rules": Array [
            "
  display: flex;
  justify-content: center;
  box-sizing: border-box;
",
          ],
        },
        "displayName": "styled.div",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-htpNat",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
